### DISCUSSION TITLE
Why does the self-supervised pre-training tutorial create two cropped views of the input image?

### DISCUSSION THREAD

**Discussion by ArjunNarayanan** [Project-MONAI/MONAI#6659] [Q&A]
Hi,

I'm taking a look at the [self-supervised pre-training tutorial](https://github.com/Project-MONAI/tutorials/blob/main/self_supervised_pretraining/vit_unetr_ssl/ssl_train.ipynb). I was wondering about the data-augmentation in the transform pipeline. Specifically, the tutorial samples two cropped views of the input image,

```
RandSpatialCropSamplesd(keys=["image"], roi_size=(96, 96, 96), random_size=False, num_samples=2)
```

I was wondering what is the purpose of this? It seems like this effectively doubles the batch-size where every subsequent image in the batch is drawn from the same input image. Thanks for any insights you can share on this!

**Comment by tangy5** [comment]
This arg can be different, if GPU mem allows, it can be set to larger num samples or 1, but remember the contrastive learning might be impacted by the num sample and batch size. 

Thank you.

**Reply by ArjunNarayanan** [reply]
I see. But by doing it this way, won't the contrastive loss treat these two randomly cropped samples as negative samples?
